packages/tpetra/core/src/Tpetra_Details_WrappedDualView.hpp

```diff
@@ -597,9 +611,10 @@ class WrappedDualView {
     // check use counts and take care of sync/modify for the underlying DualView
     //
     // The logic is this:
-    // 1) For non-CUDA archtectures where there the host/device pointers are aliased
+    // 1) If WrappedDualView tracking is disabled, then never take the sync path.
+    // 2) For non-CUDA archtectures where there the host/device pointers are aliased
     // we don't need the "sync path."
-    // 2) For CUDA, we always need the "sync path" if we're using the CudaUVMSpace (we need to make sure
+    // 3) For CUDA, we always need the "sync path" if we're using the CudaUVMSpace (we need to make sure
```

Comment:
* For documentation purposes, might be good to move this documentation such that Doxygen can pick it as the description for `needsSyncPath`?
* I am not sure to understand why these sentences are related to `Cuda` only. Isn't the WDV also used for the `HIP` case?

Reply:
That is true, this does apply to any GPU-like (heterogeneous memory) backend. Kokkos also recently added the generalization of CudaUVMSpace, called "SharedSpace", for Cuda, HIP and SYCL.

Reply:
@romintomasetti I updated the comments like you suggested. I can build the doxygen HTML locally and it looks fine now, but only if I mark the function "public:" (I guess doxygen will not run for private functions by default). But if we build docs for developers in the future, it will work here now.